test/e2e/scheduling/priorities.go

```diff
@@ -430,17 +455,46 @@ func createBalancedPodForNodes(f *framework.Framework, cs clientset.Interface, n
 	var maxCPUFraction, maxMemFraction float64 = ratio, ratio
 	var cpuFractionMap = make(map[string]float64)
 	var memFractionMap = make(map[string]float64)
+
+	var highestUtilizedNode v1.Node
 	for _, node := range nodes {
-		cpuFraction, memFraction := computeCPUMemFraction(cs, node, requestedResource)
+		cpuFraction, memFraction, _, _ := computeCPUMemFraction(cs, node, requestedResource)
 		cpuFractionMap[node.Name] = cpuFraction
 		memFractionMap[node.Name] = memFraction
 		if cpuFraction > maxCPUFraction {
 			maxCPUFraction = cpuFraction
 		}
 		if memFraction > maxMemFraction {
 			maxMemFraction = memFraction
+			highestUtilizedNode = node
 		}
 	}
+	// first, create a pod of size "crioMinMemLimit" on the most utilized node and recalculate the max
+	// this ensures that all of the following "balancePods" set at least the minimum memory limit
+	minPodConfig := &pausePodConfig{
```

Comment:
Instead of creating a new pod that will increase `maxMemFraction`, would it be sufficient to just add the min memory requirement to `podConfig.Resources.Limits` instead? E.g.
```
needCreateResource[v1.ResourceMemory] = *resource.NewQuantity(int64((ratio-memFraction)*float64(memAllocatableVal) + crioMinMemLimit), resource.BinarySI)
```

Reply:
That's a good point, I could just add that amount to every pod and it's essentially the same as creating a pod with that amount and recalculating the diff (what I'm doing now). Much easier to understand too, will update to that.

I think that also negates the need for what I'm doing above, to get the highest consumed node, right? The goal is just to make sure every pod requests at least the minimum, so adding the minimum to every pod's request will accomplish that.

Reply:
Yeah, no need to find the highest consumed node anymore.

Reply:
Updated. We could probably optimize this (in cases where pods don't need to add the entire `minMemLimit` in order to reach that value), but it would at most save a couple MB of request in a serial test anyway. I prefer the readability of "We add this much to all of them so they keep the same ratios while ensuring they request a minimum amount"